provider/pkg/provider/yaml/v2/yaml.go

```diff
@@ -0,0 +1,259 @@
+	SkipAwait      bool
+}
+
+func ParseDecodeYamlFiles(ctx *pulumi.Context, args *ParseArgs, glob bool, clientSet *clients.DynamicClientSet, opts ...pulumi.ResourceOption,
+) (pulumi.ArrayOutput, error) {
+
+	// Start with the provided objects and YAML arrays, if any, and we'll append to them.
+	objs := args.Objects
+	yamls := args.YAML
+
+	// Start by gathering any other YAML from files provided.
+	for _, file := range args.Files {
+		// Read the raw YAML file(s) specified in the input file parameter. It might be a URL or a file path.
+		var yaml []byte
+		u, err := url.Parse(file)
+		if err == nil && u.IsAbs() {
+			// If the string looks like a URL, in that it begins with a scheme, fetch it over the network.
+			resp, err := http.Get(file)
+			if err != nil {
+				return pulumi.ArrayOutput{}, errors.Wrapf(err, "fetching YAML over network")
+			}
+			defer resp.Body.Close()
+			yaml, err = io.ReadAll(resp.Body)
+			if err != nil {
+				return pulumi.ArrayOutput{}, errors.Wrapf(err, "reading YAML over network")
+			}
+			yamls = append(yamls, string(yaml))
+		} else {
+			// Otherwise, assume this is a path to a file on disk. If globbing is enabled, we might have
+			// multiple files -- otherwise just read a singular file.
+			var files []string
+			if glob {
+				files, err = filepath.Glob(file)
+				if err != nil {
+					return pulumi.ArrayOutput{}, errors.Wrapf(err, "expanding glob")
+				}
+			} else {
+				files = []string{file}
+			}
+			for _, f := range files {
+				yaml, err = os.ReadFile(f)
+				if err != nil {
+					return pulumi.ArrayOutput{}, errors.Wrapf(err, "reading YAML file from disk")
+				}
+				yamls = append(yamls, string(yaml))
+			}
+		}
+	}
+
+	// Next parse all YAML documents into objects.
+	for _, yaml := range yamls {
+		// Parse the resulting YAML bytes and turn them into raw Kubernetes objects.
+		dec, err := yamlDecode(yaml, clientSet)
+		if err != nil {
+			return pulumi.ArrayOutput{}, errors.Wrapf(err, "decoding YAML")
+		}
+		objs = append(objs, dec...)
+	}
+
+	// Now process the resulting list of Kubernetes objects.
+	return ParseYamlObjects(ctx, objs, args.ResourcePrefix, args.SkipAwait, opts...)
+}
+
+// yamlDecode decodes a YAML string into object structures.
+func yamlDecode(text string, _ *clients.DynamicClientSet) ([]map[string]any, error) {
+	var resources []unstructured.Unstructured
+
+	dec := yaml.NewYAMLOrJSONDecoder(io.NopCloser(strings.NewReader(text)), 128)
+	for {
+		var value map[string]any
+		if err := dec.Decode(&value); err != nil {
+			if err == io.EOF {
+				break
+			}
+			return nil, err
+		}
+		resource := unstructured.Unstructured{Object: value}
+
+		// Sometimes manifests include empty resources, so skip these.
+		if len(resource.GetKind()) == 0 || len(resource.GetAPIVersion()) == 0 {
+			continue
+		}
+		resources = append(resources, resource)
+	}
+
+	result := make([]map[string]any, 0, len(resources))
+	for _, resource := range resources {
+		result = append(result, resource.Object)
```

Comment:
Since we already know the length of resources, let's assign by index instead of appending to be more performant:
```
result := make([]map[string]any, len(resources))
for idx, resource := range resources {
    result[idx] = resource.Object
}
```